flair/embeddings/token.py

```diff
@@ -914,154 +909,175 @@ def _add_embeddings_to_sentences(self, sentences: List[Sentence]):
 
-        # find longest sentence in batch
-        longest_sequence_in_batch: int = len(max(subtokenized_sentences, key=len))
+        # get sentence as list of subtoken ids
+        subtoken_ids_sentence = self.tokenizer.convert_tokens_to_ids(subtokenized_sentence)
 
-        total_sentence_parts = sum(sentence_parts_lengths)
-        # initialize batch tensors and mask
-        input_ids = torch.zeros(
-            [total_sentence_parts, longest_sequence_in_batch],
-            dtype=torch.long,
-            device=flair.device,
-        )
-        mask = torch.zeros(
-            [total_sentence_parts, longest_sequence_in_batch],
-            dtype=torch.long,
-            device=flair.device,
-        )
-        for s_id, sentence in enumerate(subtokenized_sentences):
-            sequence_length = len(sentence)
-            input_ids[s_id][:sequence_length] = sentence
-            mask[s_id][:sequence_length] = torch.ones(sequence_length)
-        
-        # put encoded batch through transformer model to get all hidden states of all encoder layers
-        if type(self.tokenizer) == TransfoXLTokenizer:
-            hidden_states = self.model(input_ids)[-1]
-        else:
-            hidden_states = self.model(input_ids, attention_mask=mask)[-1]   # make the tuple a tensor; makes working with it easier.
+        # if sentence is too long, will be split into multiple parts
+        sentence_splits = []
+        while subtoken_ids_sentence:
+            encoded_inputs = self.tokenizer.encode_plus(subtoken_ids_sentence,
+                                                        max_length=self.max_subtokens_sequence_length,
+                                                        stride=self.stride,
+                                                        return_overflowing_tokens=self.allow_long_sentences,
+                                                        truncation=True,
+                                                        )
 
-        hidden_states = torch.stack(hidden_states)
+            sentence_splits.append(torch.tensor(encoded_inputs['input_ids'], dtype=torch.long))
 
-        sentence_idx_offset = 0
+            if 'overflowing_tokens' in encoded_inputs:
+                subtoken_ids_sentence = encoded_inputs['overflowing_tokens']
+            else:
+                subtoken_ids_sentence = None
 
         # gradients are enabled if fine-tuning is enabled
         gradient_context = torch.enable_grad() if (self.fine_tune and self.training) else torch.no_grad()
-
         with gradient_context:
 
-            # iterate over all subtokenized sentences
-            for sentence_idx, (sentence, subtoken_lengths, nr_sentence_parts) in enumerate(zip(sentences, subtokenized_sentences_token_lengths, sentence_parts_lengths)):
+            # embed each sentence split
+            hidden_states_of_all_splits = []
+            for split_number, sentence_split in enumerate(sentence_splits):
```

Comment:
`split_number` is currently unused in this context :)

Reply:
thanks, removed!